front/lib/api/assistant/actions/retrieval.ts

```diff
@@ -190,107 +183,22 @@ export async function retrievalActionSpecification(
-export async function generateRetrievalParams(
+// Generates the action specification for generation of rawInputs passed to `runRetrieval`.
+export async function generateRetrievalSpecification(
   auth: Authenticator,
-  configuration: AgentConfigurationType,
-  conversation: ConversationType,
-  userMessage: UserMessageType
-): Promise<
-  Result<
-    {
-      query: string | null;
-      relativeTimeFrame: TimeFrame | null;
-      topK: number | "auto";
-    },
-    Error
-  >
-> {
-  const actionConfig = deprecatedGetFirstActionConfiguration(configuration);
-
-  if (!isRetrievalConfiguration(actionConfig)) {
-    throw new Error(
-      "Unexpected action configuration received in `generateRetrievalParams`"
-    );
-  }
-
-  let query: string | null = null;
-  let relativeTimeFrame: TimeFrame | null = null;
-
-  if (
-    actionConfig.relativeTimeFrame !== "none" &&
-    actionConfig.relativeTimeFrame !== "auto"
-  ) {
-    relativeTimeFrame = actionConfig.relativeTimeFrame;
+  {
+    actionConfiguration,
+  }: {
+    actionConfiguration: RetrievalConfigurationType;
   }
-
-  const spec = await retrievalActionSpecification(actionConfig);
-
-  if (spec.inputs.length > 0) {
-    const now = Date.now();
-
-    const rawInputsRes = await generateActionInputs(
-      auth,
-      configuration,
-      spec,
-      conversation,
-      userMessage
-    );
-
-    if (rawInputsRes.isOk()) {
-      const rawInputs = rawInputsRes.value;
-
-      logger.info(
-        {
-          workspaceId: conversation.owner.sId,
-          conversationId: conversation.sId,
-          elapsed: Date.now() - now,
-        },
-        "[ASSISTANT_TRACE] Retrieval action inputs generation"
-      );
-
-      if (actionConfig.query === "auto") {
-        if (!rawInputs.query || typeof rawInputs.query !== "string") {
-          return new Err(
-            new Error("Failed to generate a valid retrieval query.")
-          );
-        }
-        query = rawInputs.query as string;
-      }
-
-      if (actionConfig.relativeTimeFrame === "auto") {
-        if (
-          rawInputs.relativeTimeFrame &&
-          typeof rawInputs.relativeTimeFrame === "string"
-        ) {
-          relativeTimeFrame = parseTimeFrame(rawInputs.relativeTimeFrame);
-        }
-      }
-    } else {
-      logger.error(
-        {
-          workspaceId: conversation.owner.sId,
-          conversationId: conversation.sId,
-          elapsed: Date.now() - now,
-          error: rawInputsRes.error,
-        },
-        "Error generating retrieval action inputs"
-      );
-
-      // We fail the rerieval only if we had to generate a query but failed to do so, if the
-      // relativeTimeFrame failed, we'll just use `null`.
-      if (actionConfig.query === "auto") {
-        return rawInputsRes;
-      }
-    }
+): Promise<Result<AgentActionSpecification, Error>> {
+  const owner = auth.workspace();
+  if (!owner) {
+    throw new Error("Unexpected unauthenticated call to `runRetrieval`");
```

Comment:
Why not return an Error since this function returns a Result?

Reply:
Here if we don't have an owner it's a 500, no point in returning an error IMO

Reply:
Yep this is a perfect case for 500 there is no world in which we would act specifically on this error in any meaningful way other than a 500

Reply:
Ok, I guess it's an issue with our auth object then, cause IMHO we should not have to have so many of those checks everywhere. Only the front-line functions should.

Reply:
could have sub-types where owner is not optional